zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionAdminAccess.java

```diff
@@ -183,21 +183,8 @@ public ActorFuture<Void> banInstance(final long processInstanceKey) {
     concurrencyControl.run(
         () -> {
           try {
-            adminControl
-                .getLogStream()
-                .newLogStreamWriter()
-                .onComplete(
-                    (writer, error) -> {
-                      if (error != null) {
-                        LOG.error(
-                            "Could not retrieve writer to write error record for process instance.",
-                            error);
-                        future.completeExceptionally(error);
-                        return;
-                      }
-
-                      writeErrorEventAndBanInstance(processInstanceKey, writer, future);
-                    });
+            final var logStreamWriter = adminControl.getLogStream().newLogStreamWriter();
```

Comment:
:question: Previous behavior was to fail the future if the writer cannot be obtained (e.g. log stream is closed). Now, we will throw an exception here, which may end up failing the concurrency control (if it's an actor)? Or what happens here?

Reply:
Caught be the surrounding try-catch block and handled gracefully.

Reply:
Oh god, I thought the try catch was part of the removed code :see_no_evil: My bad